src/accessory.ts

```diff
@@ -92,23 +95,42 @@ class PurpleAirSensor implements AccessoryPlugin {
     this.update();
   }
 
-  update() {
-    const url = 'https://www.purpleair.com/json?show=' + this.sensor;
+  async update() {
+    const url = 'https://www.purpleair.com/json';
+
+    const axiosInstance = axios.create();
+
+    axiosInstance.interceptors.request.use((request) => {
+      // write down your request intercept.
+      return AxiosLogger.requestLogger(request, {
+        logger: this.log
+      });
+    });
 
     if (this.lastReading !== undefined && this.lastReading.updateTimeMs > Date.now() - PurpleAirSensor.MIN_UPDATE_INTERVAL_MS) {
       this.log(`Skipping a fetch because the last update was ${Date.now() - this.lastReading.updateTimeMs} ms ago`);
-    } else {
-      this.log(`Fetching ${url}`);
-
-      axios.get(url).then(resp => {
-        this.lastReading = parsePurpleAirJson(resp.data, this.averages, this.conversion);
-        this.log(`Received new sensor reading ${this.lastReading}`);
-        this.updateHomeKit(this.aqiInsteadOfDensity);
-      }).catch(err => {
-        this.logger.error(`Fetching ${url}: ${err}`);
-        this.lastReading = undefined;
-        this.updateHomeKit(this.aqiInsteadOfDensity);
-      });
+      return
+    } 
+
+    try {
+      const resp = await axiosInstance.get(url, {
```

Comment:
Maybe this comment that shows my ignorance in TypeScript/nodejs. How does this await work? Does it block the thread or worse the entire process?

Reply:
I believe it will only block the thread. In practice it will behave identical to the previous then/catch implementation, I'm fairly confident.